tests/node_info_test.rs

```diff
@@ -0,0 +1,124 @@
+#[path = "../eel/tests/setup/mod.rs"]
+mod setup;
+
+#[cfg(feature = "nigiri")]
+mod node_info_test {
+    use crate::setup::nigiri;
+
+    use uniffi_lipalightninglib::CallbackError;
+    use uniffi_lipalightninglib::Config;
+    use uniffi_lipalightninglib::EventsCallback;
+    use uniffi_lipalightninglib::LightningNode;
+    use uniffi_lipalightninglib::Network;
+
+    use bitcoin::secp256k1::PublicKey;
+    use core::time::Duration;
+    use eel::errors::RuntimeErrorCode;
+    use log::info;
+    use serial_test::file_serial;
+    use std::fs;
+    use std::thread::sleep;
+
+    pub type CallbackResult<T> = std::result::Result<T, CallbackError>;
+    type Result<T> = std::result::Result<T, perro::Error<RuntimeErrorCode>>;
+
+    #[test]
+    #[file_serial(key, "/tmp/3l-int-tests-lock")]
+    fn test_get_node_info() {
+        nigiri::setup_environment_with_lsp();
+
+        let node = NodeHandle::new().start().unwrap();
+        let node_info = node.get_node_info();
+
+        assert!(
+            PublicKey::from_slice(&*node_info.node_pubkey).is_ok(),
+            "Node public key is not valid"
+        );
+        assert!(
+            node_info.channels_info.num_channels >= node_info.channels_info.num_usable_channels,
+            "Number of channels must be greater or equal to number of usable channels"
+        );
+        assert!(
+            node_info.channels_info.local_balance_msat < 21_000_000 * 100_000_000,
+            "Node must not hold more than 21 million BTC"
+        );
+    }
+
+    #[allow(dead_code)]
+    pub struct NodeHandle {
+        config: Config,
+    }
+
+    #[allow(dead_code)]
+    impl NodeHandle {
+        pub fn new() -> Self {
+            let _ = fs::remove_dir_all(".3l_local_test");
+            fs::create_dir(".3l_local_test").unwrap();
+
+            let seed: [u8; 64] = [0; 64];
+
+            let config = Config {
+                network: Network::Regtest,
+                seed: seed.to_vec(),
+                esplora_api_url: "http://localhost:30000".to_string(),
+                rgs_url: "http://localhost:8080/snapshot/".to_string(),
+                lsp_url: "http://127.0.0.1:6666".to_string(),
+                lsp_token: "iQUvOsdk4ognKshZB/CKN2vScksLhW8i13vTO+8SPvcyWJ+fHi8OLgUEvW1N3k2l"
+                    .to_string(),
+                local_persistence_path: ".3l_local_test".to_string(),
+            };
+
+            NodeHandle { config }
+        }
+
+        pub fn start(&self) -> Result<LightningNode> {
+            let events_handler = PrintEventsHandler {};
+            let node = LightningNode::new(self.config.clone(), Box::new(events_handler));
+
+            // Wait for the the P2P background task to connect to the LSP
+            sleep(Duration::from_millis(1500));
+
+            node
+        }
+    }
+
+    pub struct PrintEventsHandler {}
```

Comment:
I think you can also get this from `eel` using the `#[path = ""]` approach.

Reply:
Yes! :muscle: DOWN WITH DUPLICATE CODE

Reply:
Looks like this cannot be easily done, because the `PrintEventsHandler` in `eel` implements `EventHandler` while this `PrintEventsHandler` here in 3L is required to implement `EventsCallback`, which is being wrapped into an `EventHandler` within the 3L library.

Interestingly enough both `EventHandler` and `EventsCallback` both define the same methods. Not sure why the encapsulation but I think it has to do with error mapping.

Reply:
Yes, that's the reason for the encapsulation. Well.. seems like we can't remove the duplicate code in this case 😢